### ISSUE TITLE
Incompatibility with SQLAlchemy 1.3

### ISSUE THREAD

**Issue by gracinet** [AnyBlok/AnyBlok#83]
Noticed this with the Travis builds of Anyblok / WMS Base falling. [Restricting to `SQLAlchemy<1.3`](https://github.com/AnyBlok/anyblok_wms_base/commit/1767bd802d88e05daf49242945a41256d1bff95c) does indeed fix them.

I did not investigate further, but I'd be surprised if it were specific to AWB (Anyblok's build seems also to be currently broken on master).

```
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok_nose/plugins.py", line 106, in load_registry
    registry = RegistryManager.get(db_name)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/registry.py", line 136, in get
    db_name, loadwithoutmigration=loadwithoutmigration, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/registry.py", line 434, in __init__
    self.load()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/logging.py", line 92, in f
    return function(*args, **kwargs)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/registry.py", line 946, in load
    raise e
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/registry.py", line 942, in load
    blok2install) or mustreload
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/registry.py", line 1069, in apply_model_schema_on_table
    self.migration.auto_upgrade_database()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/migration.py", line 985, in auto_upgrade_database
    report = self.detect_changed()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/migration.py", line 996, in detect_changed
    return MigrationReport(self, diff)
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/migration.py", line 292, in __init__
    if fnct(diff):
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/migration.py", line 179, in init_remove_ck
    self.raise_if_withoutautomigration()
  File "/home/travis/virtualenv/python3.5.6/lib/python3.5/site-packages/anyblok/migration.py", line 42, in raise_if_withoutautomigration
    raise MigrationException("The metadata and the base structue are "
anyblok.migration.MigrationException: The metadata and the base structue are different, or this difference is forbidden in 'no auto migration' mode
```

**Comment by jssuzanne** [comment]
Hi, Georges.

I saw it, I fixed it in the branch pytest, can you confirm that this branch fix it. I will pick it in master and release AnyBlok.

The problem come from the naming convention, since SQLAlchemy 1.3 the name of check constrainte is truncated and finished with a hash value. 

the comparison method does not know the hash and can not determine that the truncated name and the full name are the same. What causes the destruction and recreation of the constraint

**Comment by gracinet** [comment]
Took me a while to get back to testing conditions, just gave it a new try: it seems to be fixed in 0.21.3, namely I could lauch all tests of Anyblok / Wms Base (AWB) successfully with SQLAlchemy 1.3.1

For the record, it's still there for me in branch pytest.

@jssuzanne, what would be your suggestion with respect to requirements for AWB ? Should I simply require AnyBlok >= 1.23.1 ? Would I *need* then to insist on SQLAlchemy >= 1.3 ?

**Comment by jssuzanne** [comment]
@gracinet, I think that should be in the requirement of AnyBlok 0.21.3 and upper.
